app/scripts/background.js

```diff
@@ -868,20 +872,13 @@ export function setupController(
   };
 
   /**
-   * A runtime.Port object, as provided by the browser:
-   *
-   * @see https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/Port
-   * @typedef Port
-   * @type Object
-   */
-
-  /**
-   * Connects a Port to the MetaMask controller via a multiplexed duplex stream.
+   * Connects a WindowPostMessage Port to the MetaMask controller.
    * This method identifies trusted (MetaMask) interfaces, and connects them differently from untrusted (web pages).
    *
-   * @param {Port} remotePort - The port provided by a new context.
+   * @callback ConnectWindowPostMessage
```

Comment:
I don't think adding `@callback ` here does anything.

![image](https://github.com/user-attachments/assets/45e6f2af-d1cc-4d32-a9f7-39f383ef99e8)


I think you just need to to move the type definitions to the variables, not the assignments.

Reply:
![Screenshot 2025-04-02 at 1 19 24 PM](https://github.com/user-attachments/assets/0868ae46-33fd-45ce-860f-6901f564dfe0)

doesn't work in this direction either

Reply:
Ah, sorry. It is slightly more complex than just moving it directly to the variable. I think may need to do something like:

```javascript
/**
 * @callback ConnectWindowPostMessage
 * @param {chrome.runtime.Port} remotePort - The port provided by a new context.
 * @returns {void}
 */

/**
 * This method identifies trusted (MetaMask) interfaces, and connects them differently from untrusted (web pages).
 * @type {ConnectWindowPostMessage}
 */
let connectWindowPostMessage;
```

Sorry for that misguidance!

Reply:
![Screenshot 2025-04-04 at 12 32 44 PM](https://github.com/user-attachments/assets/e180bdbc-ceb4-4778-ac1c-e2aa85bebd3b)

looks like that did it!